### PR TITLE
feat: Embed rrweb player with attachment

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+
+import AsyncComponent from 'app/components/asyncComponent';
+import {Panel, PanelBody, PanelItem} from 'app/components/panels';
+import {t} from 'app/locale';
+
+// TODO: lazy load this
+import rrwebPlayer from 'rrweb-player';
```
Comment (dcramer) [inline src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx]: would like to figure this out before shipping

**Comment by dcramer** [comment]
idc about correctness here, only that it doesnt slow down or break existing sentry

File: src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
+import PropTypes from 'prop-types';
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx]: Ca nyou write as typescript?

File: src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import React, {Component} from 'react';
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx]: Typically we just use `React.Component`, no need to import `Component` directly

File: src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+class Replayer extends Component {
+  static propTypes = {
+    url: PropTypes.string.isRequired,
+  };
+
+  constructor(...params) {
+    super(...params);
+    this.ref = React.createRef();
+  }
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx]: You can just suck `ref` out of the constructor as a instance property.

```
class ... {
   ref = React.createRef();
}
```

File: src/sentry/static/sentry/app/components/events/rrwebReplayer.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  static propTypes = {
+    url: PropTypes.string.isRequired,
+  };
+
+  constructor(...params) {
+    super(...params);
+    this.ref = React.createRef();
+  }
+
+  componentDidMount() {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/events/rrwebReplayer.jsx]: Can be written

```jsx
async componentDidMount() {
  const resp = await fetch(this.props.url)
  const data = await resp.json()

  // eslint-disable-next-line no-new
  new rrwebPlayer({
    target: this.ref.current,
    autoplay: false,
    data: {...data},
  });
}
```

File: src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+export default class RRWebIntegration extends AsyncComponent {
+  static propTypes = {
+    ...AsyncComponent.propTypes,
+    event: PropTypes.object.isRequired,
+    orgId: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+  };
+
+  componentDidMount() {
+    this.replayRef = React.createRef();
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx]: Looks unused

File: src/sentry/static/sentry/app/components/events/rrwebReplayer.jsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+export default class RRWebReplayer extends Component {
+  static propTypes = {
+    url: PropTypes.string.isRequired,
+  };
+
+  constructor(...params) {
+    super(...params);
+    this.ref = React.createRef();
+  }
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/events/rrwebReplayer.jsx]: I left this comment elsewhere, but `ref` can just be an [instance field](https://2ality.com/2017/07/class-fields.html#instance-fields).

```tsx
export default class RRWebReplayer extends Component {
   ref = React.createRef();
   ...
}

**Comment by dcramer** [comment]
ok approve my shit i fixed your nitpicks

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx (hunk 2 above)
Comment (scefali) [reply src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx]: @dcramer 

> oh wait
> nvm expert now
> dethroning evan

And you won't even write this component in typescript!

Comment (dcramer) [reply src/sentry/static/sentry/app/components/events/rrwebIntegration.jsx]: https://github.com/getsentry/sentry-rrweb/blob/master/src/index.ts pro

**Pull request merged**